File: bridge/c-sharp/Message.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Bridge
{
    // UML diyagramındaki Abstraction kısmına denk gelmektedir.
    // Implementor (IMessageSender) arayüzünün referansını tutmaktadır.
    public abstract class Message
    {
        protected IMessageSender messageSender;

        public Message(IMessageSender messageSender)
        {
            this.messageSender = messageSender;
        }

        public abstract void Send();
    }
}

File: bridge/php/Message.php
<?php

// UML diyagramındaki Abstraction kısmına denk gelmektedir.
// Implementor (IMessageSender) arayüzünün referansını tutmaktadır.
abstract class Message
{
    protected IMessageSender $_messageSender;

    public function __construct(IMessageSender $messageSender)
    {
        $this->_messageSender = $messageSender;
    }

    abstract public function send(): void;
}
